server/tasks/import_git_repo.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import git
+import json
+import sys
+import time
+import traceback
+from girder.models.folder import Folder
+from girder.models.token import Token
+from girder.models.user import User
+from girder.utility import JsonEncoder
+from girder.plugins.jobs.constants import JobStatus
+from girder.plugins.jobs.models.job import Job
+
+from ..constants import InstanceStatus, TaleStatus
+from ..models.instance import Instance
+from ..models.tale import Tale
+
+
+def run(job):
+    jobModel = Job()
+    jobModel.updateJob(job, status=JobStatus.RUNNING)
+
+    url, = job["args"]
+    if "@" in url:
+        repo_url, branch = url.split("@")
+    else:
+        repo_url = url
+        branch = "master"
+
+    user = User().load(job["userId"], force=True)
+    tale = Tale().load(job["kwargs"]["taleId"], user=user)
+    spawn = job["kwargs"]["spawn"]
+    change_status = job["kwargs"].get("change_status", True)
+    token = Token().createToken(user=user, days=0.5)
+
+    progressTotal = 1 + int(spawn)
+    progressCurrent = 0
+
+    try:
+        # 1. Checkout the git repo
+        jobModel.updateJob(
+            job,
+            status=JobStatus.RUNNING,
+            progressTotal=progressTotal,
+            progressCurrent=progressCurrent,
+            progressMessage="Cloning the git repo",
+        )
+
+        workspace = Folder().load(tale["workspaceId"], force=True)
+        try:
+            repo = git.Repo.init(workspace["fsPath"])
```

Comment:
Would it make sense to check for the existences of a git repo prior to initing?  If `.git` exists, I think we should just abandon ship ("It appears that you have an existing git repo... we don't support that. Use command line tools."). And, if `.git` doesn't exist then we know we can remove it on fetch failure.